homeassistant/components/camera/__init__.py

```diff
@@ -275,6 +275,12 @@ def handle_async_mjpeg_stream(self, request):
 
         last_image = None
 
+        interval = .5  # default stream framerate: 2 fps
+        min_interval = .05  # max stream framerate: 20 fps
+
+        if request.query.get('interval'):
+            interval = max(float(request.query.get('interval')), min_interval)
```

Comment:
Let's please not use max here, instead let's raise an exception, render a 500 or something like that to indicate wrong input.

I'm sure some users will be confused when their interval isn't accepted by the server. Some users will definitely try to get 30fps.

Also, I feel like the request parameter could also be called FPS, since that's more the standard of cameras.

Reply:
Makes sense. And then just use `1/fps` as parameter for `asyncio.sleep()`

Reply:
Yes, we could raise a `403 Forbidden` with a description of the reason.
In the frontend we would limit what users can select to valid framerate.

Reply:
Hmmm, Maybe fps is not too good after all: If fetching the latest image + sending it takes a long time, we would always have *a little bit less* than the fps. That could also lead to confusion.

So do we just leave that limitation on there or track time by ourselves to achieve "true" fps...?

Reply:
I'd think we can just leave the limitation. It's not a video player.... achieving 'true fps' is not really the point.

Reply:
I believe INTERVAL would be a better description of the behaviour here.
Users who will figure out how to get to the stream, will be able to understand what an interval is and how that relates to 'fps' I guess.

Reply:
Yes, I agree.